apps/testing/src/main/java/com/akto/testing/Main.java

```diff
@@ -453,6 +457,34 @@ public void run() {
 
                 loggerMaker.infoAndAddToDb("Tests completed in " + (Context.now() - start) + " seconds for account: " + accountId, LogDb.TESTING);
 
+                /*
+                 * In case the testing run results start overflowing
+                 * due to being a capped collection,
+                 * we will start deleting non-vulnerable results,
+                 * since we cap on size as well as number of documents.
+                 */
+                deleteScheduler.execute(() -> {
+                    Context.accountId.set(accountId);
+                    try {
+                        /*
+                         * Since this is a heavy job,
+                         * running it for only one account on one instance at a time.
+                         */
+                        boolean dibs = Cluster.callDibs(Cluster.DELETE_TESTING_RUN_RESULTS, 60 * 60, 60);
```

Comment:
if dibs is not acquired, then we will never trim the collection. Need to handle that.

Reply:
acquire dibs post knowing that we have reached a limit